extensions/control-plane/provision/provision-gcs/src/main/java/org/eclipse/edc/connector/provision/gcp/GcsProvisioner.java

```diff
@@ -76,9 +82,17 @@ public CompletableFuture<StatusResult<ProvisionResponse>> provision(
         try {
             var bucket = storageService.getOrCreateBucket(bucketName, bucketLocation);
 
-            // TODO use service account from transfer request, in case defined.
-            var serviceAccount = createServiceAccount(processId, bucketName);
-            var token = createBucketAccessToken(bucket, serviceAccount);
+            GcpServiceAccount serviceAccount = null;
+            GcpAccessToken token = null;
+
+            var serviceAccountName = getServiceAccountName(resourceDefinition);
+            if (serviceAccountName != null && !serviceAccountName.isEmpty()) {
```

Comment:
I think `serviceAccountName` can never be null (please look at `GcpConfiguration`). 

unrelated: I think that's not a good approach to use empty string as default value, it would be better to leave `null` for not-configured settings, so a simple not-null check will already permit to say if that configuration exists or not.

Reply:
Updated GcpConfiguration, not returning null now